File: app/schemas/post.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.db.enums import PostType


class PostCreateRequest(BaseModel):
    title: str
    body: str = Field(description="Rich text HTML body. Server sanitizes before persistence.")
    type: PostType


class PostResponse(BaseModel):
    id: uuid.UUID
    title: str
    body: str = Field(description="Sanitized rich text HTML body.")
    type: PostType
    author_id: uuid.UUID
    created_at: datetime

    class Config:
        from_attributes = True


class PostAttachmentUploadResponse(BaseModel):
    url: str
    href: str
    filename: str
    content_type: str
    filesize: int

File: frontend/src/types/post.ts
export type PostType = "article" | "case_study";

export interface Post {
  id: string;
  title: string;
  body: string;
  type: PostType;
  author_id: string;
  created_at: string;
  author_name?: string;
  author?: {
    username?: string;
  };
}

export interface PostCreateRequest {
  title: string;
  body: string;
  type: PostType;
}

export interface PostAttachmentUploadResponse {
  url: string;
  href: string;
  filename: string;
  content_type: string;
  filesize: number;
}
